lib/network.go

```diff
@@ -1827,34 +1846,136 @@ func (msg *MsgDeSoGetAddr) GetMsgType() MsgType {
 // VERACK Message
 // ==================================================================
 
-// VERACK messages have no payload.
+type VerackVersion uint64
+
+const (
+	VerackVersion0 VerackVersion = 0
+	VerackVersion1 VerackVersion = 1
+)
+
+func (vv VerackVersion) ToUint64() uint64 {
+	return uint64(vv)
+}
+
 type MsgDeSoVerack struct {
+	Version VerackVersion
+
 	// A verack message must contain the nonce the peer received in the
 	// initial version message. This ensures the peer that is communicating
 	// with us actually controls the address she says she does similar to
 	// "SYN Cookie" DDOS protection.
-	Nonce uint64
+	NonceReceived uint64
+	NonceSent     uint64
+	TstampMicro   uint64
```

Comment:
Nanoseconds are pointless, they overflow and are inaccurate. I don't like our choice of nanoseconds in other parts of the code, there's never a need for such granularity. In fact the timestamp in nanos is a multiple of 1000, because it's rounded to nearest microsecond.